packages/amplify-category-auth/src/index.js

```diff
@@ -312,6 +318,24 @@ async function executeAmplifyCommand(context) {
   await commandModule.run(context);
 }
 
+/**
+ * Entry point for headless commands
+ * @param {any} context The amplify context object
+ * @param {string} headlessPayload The serialized payload from the platform
+ */
+const executeAmplifyHeadlessCommand = async (context, headlessPayload) => {
+  switch (context.input.command) {
+    case 'add':
+      if (projectHasAuth(context)) return;
+      await validateAddAuthRequest(headlessPayload)
+        .then(addAuthRequestAdaptorFactory(context.amplify.getProjectConfig().frontend))
```

Comment:
nit: why `then` in async code? It would be nice to use only `async`-`await` where we can

Reply:
To me, async / await is good when you need to manipulate the result of the promise in the calling function. But if you just need to pipe it into another function (as is the case here), using then is cleaner because you don't have to have nested function calls or unnecessary variable assignments